GrpcClient/EmployeeGrpcRepository.cs

```diff
@@ -0,0 +1,286 @@
+    public async Task<Entities.Employee> AddAsync(Entities.Employee employee)
+    {
+        using var channel = GrpcChannel.ForAddress(_grpcAddress); 
+        //create client for employee entity, "EmployeeClient" method is auto-genereted               
+        var client = new Employee.EmployeeClient(channel);
+        //reply - "AddSingleEmployee(newEmployeeDto)" comes from .proto file + need to convert the og-DTO to .proto-DTO
+        try
+        {
+            var reply = await client.AddSingleEmployeeAsync(new NewEmployeeDTO
+            {
+                FirstName = employee.FirstName,
+                LastName = employee.LastName,
+                WorkingNumber = UInt32.CreateChecked(employee.WorkingNumber),
+                Email = employee.Email,
+                Password = employee.Password
+            });
+            //converting
+            List<ShiftDTO> grpcShiftDtos = reply.AssignedShifts.Dtos.ToList();
+            List<Entities.Shift> shifts = GrpcShiftDtosToEntityShiftsList(grpcShiftDtos);
+            Entities.Employee employeeRecieved = grpcEmployeeDtoToEntityEmployee(reply, shifts);
+        
+            return employeeRecieved;
+        }
+        catch (RpcException e)
+        {
+            if (e.StatusCode == StatusCode.AlreadyExists)
+            {
+                throw new ArgumentException(e.Message);
+            }
+        }
+
+
+
+
+        return null;
+    }
+
+    public async Task<Entities.Employee> UpdateAsync(Entities.Employee employee)
+    {
+        Entities.Employee employee2 = new Entities.Employee();
+        try
+        {
+            using var channel = GrpcChannel.ForAddress(_grpcAddress); 
+            var client = new Employee.EmployeeClient(channel);
+            var reply = await client.UpdateSingleEmployeeAsync(new UpdateEmployeeDTO
+            {
+                Id = employee.Id,
+                FirstName = employee.FirstName,
+                LastName = employee.LastName,
+                WorkingNumber = UInt32.CreateChecked(employee.WorkingNumber),
+                Email = employee.Email,
+                Password = employee.Password
+            });
+            employee2 = grpcEmployeeDtoToEntityEmployee(reply,
+                GrpcShiftDtosToEntityShiftsList(reply.AssignedShifts.Dtos.ToList()));
+        }
+        catch (RpcException e)
+        {
+            if (e.StatusCode == StatusCode.NotFound)
+            {
+                throw new ArgumentException(e.Message);
+            }
+            else if (e.StatusCode == StatusCode.AlreadyExists)
+            {
+                throw new ArgumentException(e.Message);
+            }
+        }
+
+        return employee2;
+    }
+
+    public async Task DeleteAsync(long id)
+    {
+        try
+        {
+            using var channel = GrpcChannel.ForAddress(_grpcAddress); 
+            var client = new Employee.EmployeeClient(channel);
+            var reply = await client.DeleteSingleEmployeeAsync(new Id { Id_ = id });
```

Comment:
Why there is _ in Id_ ? I'm a bit confused on what is going in with (new Id { Id_ = id })

Reply:
its new Id in grpc if u hover over it and yea you are creating new one with its property Id_ and assigning id (from csharp) to it